workflow/common/validate.go

```diff
@@ -114,6 +119,17 @@ func validateInputs(tmpl *wfv1.Template) (map[string]interface{}, error) {
 	return scope, nil
 }
 
+func validateWFGlobalParams(tmpl *wfv1.Template, wfGlobalParams []wfv1.Parameter, scope map[string]interface{}) error {
+	err := VerifyUniqueNonEmptyNames(wfGlobalParams)
+	if err != nil {
+		return errors.Errorf(errors.CodeBadRequest, "workflow.spec.Arguments.parameters non unique names for %s", err.Error())
```

Comment:
VerifyUniqueNonEmptyNames() returns a partial error string which was intended to be combined to a more specific error message. But it also returns error if the name is empty, so this message can be misleading. To make it consistent with other errors, can we change this to:
```
return errors.Errorf(errors.CodeBadRequest, "'spec.arguments.parameters%s", err.Error())
```
This will produce errors like:
`spec.arguments.parameters[1].name is required`
or 
`spec.arguments.parameters[1].name 'message' is not unique`

Reply:
BTW, in the near future, we also need to validate that the name is a kubernetes object name and there aren't weird characters in names (including periods).